packages/@react-aria/dnd/src/useDrop.ts

```diff
@@ -158,30 +159,43 @@ export function useDrop(options: DropOptions): DropResult {
     e.preventDefault();
     e.stopPropagation();
 
-    if (typeof options.onDrop === 'function') {
-      let dropOperation = DROP_EFFECT_TO_DROP_OPERATION[state.dropEffect];
-      let items = readFromDataTransfer(e.dataTransfer);
-
-      let rect = (e.currentTarget as HTMLElement).getBoundingClientRect();
-      let event: DropEvent = {
-        type: 'drop',
-        x: e.clientX - rect.x,
-        y: e.clientY - rect.y,
-        items,
-        dropOperation
-      };
-
-      // Wait a frame to dispatch the drop event so that we ensure the dragend event fires first.
-      // Otherwise, if onDrop removes the original dragged element from the DOM, dragend will never be fired.
-      // This behavior is consistent across browsers, but see this issue for details:
-      // https://bugzilla.mozilla.org/show_bug.cgi?id=460801
-      setTimeout(() => {
+    let dropOperation = DROP_EFFECT_TO_DROP_OPERATION[state.dropEffect];
+    let items = readFromDataTransfer(e.dataTransfer);
+
+    let rect = (e.currentTarget as HTMLElement).getBoundingClientRect();
+    let event: DropEvent = {
+      type: 'drop',
+      x: e.clientX - rect.x,
+      y: e.clientY - rect.y,
+      items,
+      dropOperation
+    };
+    // Set drop effect in global DnD state for Chrome Android. TODO: file bug against chrome
+    // where onDragEnd always returns "none" as its drop effect.
+    setDropEffect(state.dropEffect);
+
+    // In the case where a drop happens on a non-collection drop target, track the element in which the drop was performed
+    setDroppedTarget(e.currentTarget as HTMLElement);
+
+    let dndStateSnapshot = {...getDnDState()};
+    // Wait a frame to dispatch the drop event so that we ensure the dragend event fires first.
+    // Otherwise, if onDrop removes the original dragged element from the DOM, dragend will never be fired.
+    // This behavior is consistent across browsers, but see this issue for details:
+    // https://bugzilla.mozilla.org/show_bug.cgi?id=460801
+    setTimeout(() => {
+      setDnDState(dndStateSnapshot);
```

Comment:
Alternative to snapshotting and briefly making the global DnD state inaccurate (kinda? More like our dnd flow has some extra calls due to the setTimeouts in useDrop + useDrag) is to freeze the DnD state and store that separately. Roughly implemented here: [31610ec](https://github.com/adobe/react-spectrum/commit/31610ec5b3d4e287c0ff3bf93c564e329243d78f) but felt a bit overkill/adds too much complexity with having two different DnD states to think about